core/src/main/scala/org/apache/spark/storage/BlockManagerMessages.scala

```diff
@@ -108,4 +107,6 @@ private[spark] object BlockManagerMessages {
     extends ToBlockManagerMaster
 
   case class BlockManagerHeartbeat(blockManagerId: BlockManagerId) extends ToBlockManagerMaster
+
+  case class HasCachedBlocks(executorId: String) extends ToBlockManagerMaster
```

Comment:
But I don't see it being sent anywhere? I see a handler for it but no code actually sending the message.